Detectors/ITSMFT/ITS/workflow/src/DeadMapBuilderSpec.cxx

```diff
@@ -0,0 +1,389 @@
+    outfile.cd();
+    mTreeObject->Write();
+    outfile.Close();
+  }
+  return;
+
+} // finalizeOutput
+
+//////////////////////////////////////////////////////////////////////////////
+// Main running function
+void ITSDeadMapBuilder::run(ProcessingContext& pc)
+{
+  if (mRunStopRequested) { // give up when run stop request arrived
+    return;
+  }
+
+  std::chrono::time_point<std::chrono::high_resolution_clock> start;
+  std::chrono::time_point<std::chrono::high_resolution_clock> end;
+
+  start = std::chrono::high_resolution_clock::now();
+
+  mTFCounter++;
+
+  mFirstOrbitTF = pc.services().get<o2::framework::TimingInfo>().firstTForbit;
+
+  if ((Long64_t)(mFirstOrbitTF / mTFLength) % mTFSampling != 0) {
+    return;
+  }
+
+  mStepCounter++;
+  LOG(info) << "Processing step #" << mStepCounter << " out of " << mTFCounter << " TF received. First orbit " << mFirstOrbitTF;
+
+  mLanesAlive.clear();
+  mDeadMapTF->clear();
+
+  bool newlane = false;
+
+  if (mDataSource == "digits") {
+    const auto elements = pc.inputs().get<gsl::span<o2::itsmft::Digit>>("elements");
+    const auto ROFs = pc.inputs().get<gsl::span<o2::itsmft::ROFRecord>>("ROFs");
+    for (const auto& rof : ROFs) {
+      auto elementsInTF = rof.getROFData(elements);
+      for (const auto& el : elementsInTF) {
+        short int chipID = (short int)el.getChipIndex();
+        newlane = mLanesAlive.insert(getLaneIDFromChip(chipID)).second;
+      }
+    }
+  } else if (mDataSource == "clusters") {
+    const auto elements = pc.inputs().get<gsl::span<o2::itsmft::CompClusterExt>>("elements");
+    const auto ROFs = pc.inputs().get<gsl::span<o2::itsmft::ROFRecord>>("ROFs");
+    for (const auto& rof : ROFs) {
+      auto elementsInTF = rof.getROFData(elements);
+      for (const auto& el : elementsInTF) {
+        short int chipID = (short int)el.getSensorID();
+        newlane = mLanesAlive.insert(getLaneIDFromChip(chipID)).second;
+      }
+    }
+  } else if (mDataSource == "chipsstatus") {
+    const auto elements = pc.inputs().get<std::vector<char>>("elements");
+    for (short int chipID = 0; chipID < elements.size(); chipID++) {
+      if (elements.at(chipID)) {
+        newlane = mLanesAlive.insert(getLaneIDFromChip(chipID)).second;
+      }
+    }
+  }
+
+  mLanesAlive.insert(getLaneIDFromChip(N_CHIPS));
+
+  LOG(info) << "TF contains" << mLanesAlive.size() << " active elements";
+  // filling the vector
+  std::pair<int, int> fillresult = FillMapElementITS(); // first = n dead lanes, second = n words
+
+  // filling the tree
+  mTreeObject->Fill();
+
+  end = std::chrono::high_resolution_clock::now();
+  int difference = std::chrono::duration_cast<std::chrono::microseconds>(end - start).count();
+
+  LOG(info) << "Elapsed time in TF processing: " << difference / 1000. << " ms";
+
+  return;
+}
+
+std::pair<int, int> ITSDeadMapBuilder::FillMapElementITS()
+{
+
+  if (MAP_VERSION == "1") { // Force developer to change metadata in case encoding is changed
```

Comment:
You are using fairly complicated packing which I did not fully understand, but why do you need it and why are you storing alive lanes instead of much fewer dead ones?
I would simply store the dead lane IDs in e.g. `std::vector<uint16_t> deadLanes` with the convention that if 
`(deadLanes[i]  &0x8000) == 0`  then this is a single lane, otherwise, if this is a start of the range (followed by the end of the range w/o the highest bit set). In this case you will need to store only a handful of entries per TF

Reply:
Hi @shahor02 , thanks for checking this out. 
It's saving only the dead lanes (the first loop over clusters identifies the alive lanes, the `FillMapElement` saves the IDs in between two alive IDs).
Concerning the packing, it was intended to leave as few unused bits as possible, but in the end the larger compression is given by saving ranges. So I fully agree with your simplification.
Going to push a review. Thanks.